DataStructures/DynamicStacks/UnitTestDynamicStack.java

```diff
@@ -0,0 +1,91 @@
+public class UnitTestDynamicStack{
+    public static void main(String[] args){
+        DynamicStack ds = new DynamicStack();
+        DynamicStack dswm = new DynamicStack(5);
+
+        //Testing paramaterless constructor 
+        int numberIn = 10;
+        int numberOut = 3;
+        TestDynamicStack(numberIn, numberOut, ds);
+        System.out.println("\n\n\n");
+
+
+        //Testing constructor with size limit
+        numberIn = 8;
+        numberOut = 3;
+        TestDynamicStack(numberIn, numberOut, dswm);
+        System.out.println("\n\n\n");
+
+
+        Node nd = new Node()
+            .withPayload(5);
+
+        TestWithNode(nd, 5);
+
+        Node nd2 = new Node().withPayload(3).withNext(nd);
+        TestWithNode(nd2, 3);
```

Comment:
This is checking the payload buy it's not checking that the next of nd2 is the expected value (i.e nd)

Reply:
This should now be corrected.